File: src/main/java/org/example/websitesalephone/service/cart/CartService.java
package org.example.websitesalephone.service.cart;

import org.example.websitesalephone.comon.CommonResponse;
import org.example.websitesalephone.dto.cart.CartRequest;
import org.example.websitesalephone.dto.cart.CartSearch;
import org.example.websitesalephone.dto.cart.CheckOutRequest;
import org.example.websitesalephone.dto.cart.BuyNowCartRequest;

public interface CartService {

    CommonResponse addToCart(CartRequest request);

    CommonResponse updateCartItem(CartRequest request);

    CommonResponse getCartItems(CartSearch search);

    CommonResponse checkoutCart(CheckOutRequest checkOutRequest);

    CommonResponse buyNow(BuyNowCartRequest buyNowCartRequest);
}

File: cms-user/src/service/CartService.ts
import type { AxiosResponse } from 'axios'
import api from '../api/api'
import type { ICartRequest } from '../models/CartRequest'
import type { ICheckOutRequest } from '../models/CheckOutRequest'

class CartService {
    private readonly ROOT_API = `${import.meta.env.VITE_ROOT_API}/api/cart/`

    public addToCart(request: ICartRequest): Promise<AxiosResponse> {
        return api.post(`${this.ROOT_API}add`, request)
    }

    public updateCartItem(request: ICartRequest): Promise<AxiosResponse> {
        return api.put(`${this.ROOT_API}update`, request)
    }

    public getCartItems(search: Record<string, unknown>): Promise<AxiosResponse> {
        return api.post(`${this.ROOT_API}items`, search)
    }

    public checkoutCart(checkOutRequest: ICheckOutRequest): Promise<AxiosResponse> {
        return api.post(`${this.ROOT_API}checkout`, checkOutRequest)
    }

    public buyNow(buyNowCartRequest: {
        variantId: string
        quantity: number
        addressLine: string
        paymentMethod: string
    }): Promise<AxiosResponse> {
        return api.post(`${this.ROOT_API}buy-now`, buyNowCartRequest)
    }
}

export const cartService = new CartService()
